weld/llvm.rs

```diff
@@ -1993,18 +1993,83 @@ impl LlvmGenerator {
                                                              bld_ty_str,
                                                              llvm_symbol(output)));
                                     }
-                                    Merger(_, ref op) => {
-                                        if *op != BinOpKind::Add {
-                                            return weld_err!("Merger only supports +");
-                                        }
-                                        let bld_ty_str = try!(self.llvm_type(ty));
+                                    Merger(ref elem_ty, ref op) => {
+                                        let bld_ty_str = self.llvm_type(ty)?.to_string();
+                                        let elem_type = (self.llvm_type(elem_ty)?).to_string();
                                         let bld_prefix = format!("@{}",
                                                                  bld_ty_str.replace("%", ""));
                                         let bld_tmp = ctx.var_ids.next();
                                         ctx.code.add(format!("{} = call {} {}.new()",
                                                              bld_tmp,
                                                              bld_ty_str,
                                                              bld_prefix));
+
+                                        // Generate code to initialize the builder.
+                                        let iden_elem = match *op {
+                                            BinOpKind::Add => {
+                                                match **elem_ty {
+                                                    Scalar(F32) | Scalar(F64) => "0.0".to_string(),
+                                                    _ => "0".to_string(),
+                                                }
+                                            }
+                                            BinOpKind::Multiply => {
+                                                match **elem_ty {
+                                                    Scalar(F32) | Scalar(F64) => "1.0".to_string(),
+                                                    _ => "1".to_string(),
+                                                }
+                                            }
+
+                                            _ => {
+                                                return weld_err!("Invalid merger binary op in \
+                                                                  codegen")
+                                            }
+                                        };
+
+                                        let init_elem = match *arg {
```

Comment:
It's the argument passed to `NewBuilder` above. I added a comment for this line.